opalog/term.go

```diff
@@ -111,7 +123,11 @@ func (term1 *Term) Equal(term2 *Term) bool {
 	case VAR:
 		var1 := term1.Value.(*Var)
 		var2 := term2.Value.(*Var)
-		return var1.Name == var2.Name
+		return var1.Equal(var2)
+	case REF:
+		ref1 := term1.Value.(*Ref)
+		ref2 := term2.Value.(*Ref)
+		return ref1.Equal(ref2)
```

Comment:
Still not sure what to think of the tradeoffs between having 1 struct for each of Var, Array, Object, etc. and having 1 struct for all Terms.  Seems that in both cases we'll want this kind of Term Equal Term function, at least for convenience.  I'd say we'll only want 1 Struct of memory overhead for each Term.  Right now it looks like we have 2: Term and Ref.  (Or does Go optimize away structs somehow?)  Maybe that's premature optimization, but it's also inevitable, so maybe it's best to just have 1 at the start.